src/main/java/trinity/play2learn/backend/admin/year/services/commons/YearExistService.java

```diff
@@ -0,0 +1,34 @@
+package trinity.play2learn.backend.admin.year.services.commons;
+
+import org.springframework.stereotype.Service;
+
+import trinity.play2learn.backend.admin.year.repositories.IYearRepository;
+import trinity.play2learn.backend.admin.year.services.interfaces.IYearExistService;
+
+/**
+ * Servicio para validar si ya existe un año con el mismo nombre.
+ */
+@Service
+public class YearExistService implements IYearExistService {
+
+    private IYearRepository yearRepository;
+
+    public YearExistService(IYearRepository yearRepository) {
+        this.yearRepository = yearRepository;
+    }
+
+    /**
+     * Recibe un string y valida si existe un año con ese nombre.
+     *
+     * @param String name.
+     * @return Boolean.
+     */
+    @Override
+    public boolean validate(String name) {
+        if (yearRepository.findByName(name).isPresent()) {
+            return true;
+        }
+        return false;
+    }
```

Comment:
Acá un par de cuestiones. Me parece medio innecesario crear una clase para esta validación, sobre todo cuando JPA te proporciona una manera excelente de realizar estas validaciones a través de los [Query Methods](https://docs.spring.io/spring-data/jpa/reference/jpa/query-methods.html).

Con un query method, agarrás el `YearRepository` y le definís el método de forma verbal.

```
// Dentro del YearRepository
boolean existsByName(String name);
```

Y se implementa automáticamente. Magia.

Ahora, si y sólo si preferís implementar ese método así, nunca jamás de los jamases evalúes una condición y luego devuelvas un booleano. Devolvé el booleano directamente

```suggestion
    public boolean validate(String name) {
        return yearRepository.findByName(name).isPresent();
    }
```